compiler-rt/lib/rtsan/rtsan_interceptors.cpp

```diff
@@ -461,7 +461,17 @@ INTERCEPTOR(void *, valloc, SIZE_T size) {
   return REAL(valloc)(size);
 }
 
-#if SANITIZER_INTERCEPT_ALIGNED_ALLOC
+// aligned_alloc was introduced in OSX 10.15
+// Linking will fail when using an older SDK
+#if defined(__MAC_10_15)
+// macOS 10.15 is greater than our minimal deployment target.  To ensure we
+// generate a weak reference so the dylib continues to work on older
+// systems, we need to forward declare the intercepted function as "weak
+// imports".
+SANITIZER_WEAK_IMPORT void *aligned_alloc(SIZE_T __alignment, SIZE_T __size);
+#endif // defined(__MAC_10_15)
+
+#if SANITIZER_INTERCEPT_ALIGNED_ALLOC || SANITIZER_APPLE
```

Comment:
Shouldn't we just set `SANITIZER_INTERCEPT_ALIGNED_ALLOC` to `true` for `SANITIZER_APPLE` then?

Reply:
Definitely, but this has a much bigger implications that touch almost all of the other sanitizers. The majority of them that care about allocations assume this is linux only, and don't have this WEAK_IMPORT thing.

I opened this thread to discuss bubbling this up, which I agree is the ideal end goal.

https://discourse.llvm.org/t/should-aligned-alloc-be-intercepted-on-osx/82599

Reply:
My proposal is get feedback on this change to make sure it looks good, integrate it, then follow it up with a review that makes this true for all of the sanitizers

Reply:
I meant something like this in appropriate place

```
#if SANITIZER_APPLE
#undef SANITIZER_INTERCEPT_ALIGNED_ALLOC
#define SANITIZER_INTERCEPT_ALIGNED_ALLOC 1
#endif
```